.circleci/config.yml

```diff
@@ -353,47 +355,78 @@ workflows:
+          PYTEST_ARGS: --ignore=tests/test_namespace.py --ignore=tests/test_leaks.py --ignore=tests/test_kat_all.py --numprocesses=auto
       - linux_oqs:
           <<: *require_buildcheck
           name: ubuntu-focal-clang15
           context: openquantumsafe
           CONTAINER: openquantumsafe/ci-ubuntu-focal-x86_64:latest
           CMAKE_ARGS: -DOQS_STRICT_WARNINGS=ON -DCMAKE_C_COMPILER=clang-15 -DOQS_OPT_TARGET=skylake
+          PYTEST_ARGS: --ignore=tests/test_kat_all.py
       - linux_oqs:
           <<: *require_buildcheck
           name: ubuntu-bionic-i386
           context: openquantumsafe
           CONTAINER: openquantumsafe/ci-ubuntu-bionic-i386:latest
           CMAKE_ARGS: -DCMAKE_TOOLCHAIN_FILE=../.CMake/toolchain_x86.cmake
-          PYTEST_ARGS: --ignore=tests/test_leaks.py
+          PYTEST_ARGS: --ignore=tests/test_leaks.py --ignore=tests/test_kat_all.py
       - arm_machine:
           <<: *require_buildcheck
           name: arm64
-          PYTEST_ARGS: --numprocesses=auto --maxprocesses=10
+          PYTEST_ARGS: --numprocesses=auto --maxprocesses=10 --ignore=tests/test_kat_all.py
       - macOS:
           <<: *require_buildcheck
           name: macOS-gcc11
           CMAKE_ARGS: -DCMAKE_C_COMPILER=gcc-11
+          PYTEST_ARGS: --ignore=tests/test_kat_all.py
       - macOS:
           <<: *require_buildcheck
           name: macOS-noopenssl
           CMAKE_ARGS: -DOQS_STRICT_WARNINGS=ON -DOQS_USE_OPENSSL=OFF
+          PYTEST_ARGS: --ignore=tests/test_kat_all.py
       - macOS:
           <<: *require_buildcheck
           name: macOS-shared
           CMAKE_ARGS: -DBUILD_SHARED_LIBS=ON -DOQS_DIST_BUILD=OFF -DOQS_ENABLE_KEM_CLASSIC_MCELIECE=OFF
+          PYTEST_ARGS: --ignore=tests/test_kat_all.py
 
   commit-to-main:
     when:
-      equal: [ main, << pipeline.git.branch >> ]
+      and:
+        - not:
+            equal: [ scheduled_pipeline, << pipeline.trigger_source >> ]
+        - equal: [ main, << pipeline.git.branch >> ]
     jobs:
       - trigger-downstream-ci:
           context: openquantumsafe
+
+  nistkat:
+    when:
+      and:
+        - equal: [ scheduled_pipeline, << pipeline.trigger_source >> ]
+        - equal: [ "NIST KATs", << pipeline.schedule.name >> ]
+        - equal: [ sw-full-kat, << pipeline.git.branch >> ]
+    jobs:
+      - linux_oqs:
+          name: nistkat-generic
+          context: openquantumsafe
+          CONTAINER: openquantumsafe/ci-ubuntu-focal-x86_64:latest
+          CMAKE_ARGS: -DOQS_OPT_TARGET=generic
+          PYTEST_ARGS: --numprocesses=auto -k 'test_kat_all'
+      - linux_oqs:
+          name: nistkat-avx2
+          context: openquantumsafe
+          CONTAINER: openquantumsafe/ci-ubuntu-focal-x86_64:latest
+          CMAKE_ARGS: -DOQS_DIST_BUILD=OFF
+          PYTEST_ARGS: --numprocesses=auto -k 'test_kat_all'
+      - arm_machine:
+          name: nistkat-aarch64
+          CMAKE_ARGS: -DOQS_DIST_BUILD=OFF
```

Comment:
This should then only test the "auto" config on `aarch64`. Is this what we want? I'd rather suggest a "generic"/ref-code and a "neon"/optimized run.

Reply:
Right, so the generic run would have `-DOQS_DIST_BUILD=OFF -DOQS_OPT_TARGET=generic` and the optimized run would have `-DOQS_DIST_BUILD=OFF` (with `OQS_OPT_TARGET` defaulting to `native`)?

Reply:
Sounds right -- but I'm not knowledgeable enough about ARM to authoritatively confirm this. But then again, if it works for `x64` why should it be wrong for `aarch64` :-)